src/lua/datetime.lua

```diff
@@ -1090,11 +1090,17 @@ local function datetime_set(self, obj)
         -- otherwise - use nsec, usec, or msec
         if count_usec == 0 then
             nsec = fraction * 1e9
-        else
+        elseif fraction ~= 0 then
             error('only integer values allowed in timestamp '..
                   'if nsec, usec, or msecs provided', 2)
         end
 
+        if usec ~= nil then
+            nsec = usec * 1e3
+        end
+        if msec ~= nil then
+            nsec = msec * 1e6
+        end
```

Comment:
>  But: could you check ranges for usec and msec in your patch, like Timur did in other places?

I think I can do that but it will be useless since it's checked by [2].
Let's won't introduce double-check

```lua
tarantool> ts:set{timestamp = 1630359071, msec = 1000000}
---
- error: value 1000000 of msec is out of allowed range [0, 1000]
...

tarantool> ts:set{timestamp = 1630359071, nsec = -1}
---
- error: value -1 of nsec is out of allowed range [0, 1000000000]
...

tarantool> ts:set{timestamp = 1630359071, usec = -1}
---
- error: value -1 of usec is out of allowed range [0, 1000000]
...
```